### DISCUSSION TITLE
How to open metamask for transaction confirmation in wagmi mobile wallet-connect

### DISCUSSION THREAD

**Discussion by im-VIKI** [wevm/wagmi#2030] [Q&A]
after connecting wallet, how can i open metamask wallet automatic for transaction confirmation instead of doing it mannully.

how its look after click on claim button not open the metamask wallet.
![screen shot](https://user-images.githubusercontent.com/22514919/226121314-d6756ef5-ed95-483e-94de-6ffa5698fe8c.jpeg)

**Comment by SaadNiaziDev** [comment]
You can call write function in onConnect callback of useAccount Hook

https://wagmi.sh/react/hooks/useAccount#onconnect-optional

but be carefull as wagmi reconnects on each refresh if we set AutoConnect true, so use isReconnected flag to avoid transaction poping on every refresh

`const account = useAccount({
    onConnect({ address, connector, isReconnected }) {
      if(!isReconnected){
          claim()
     }
    },
  })`
